lib/containers/bats.pm

```diff
@@ -78,6 +79,20 @@ sub install_bats {
     assert_script_run "chmod +x /usr/local/bin/bats_skip_notok";
 }
 
+sub bats_setup {
+    my $self = shift;
+
+    $self->delegate_controllers;
+    $self->remove_mounts_conf;
+    # Disable tmpfs from next boot
+    assert_script_run "systemctl mask tmp.mount";
+    # Switch to cgroup v2 if not already active
+    if (script_run("ls /sys/fs/cgroup/cgroup.controllers") != 0) {
+        add_grub_cmdline_settings("systemd.unified_cgroup_hierarchy=1", update_grub => 1);
+    }
+    $self->wait_boot();
+}
```

Comment:
I think you are missing a boot trigger in after the grub settings change

Reply:
Fixed.